broadcaster/_base.py

```diff
@@ -72,9 +72,17 @@ async def disconnect(self) -> None:
     async def _listener(self) -> None:
         while True:
             event = await self._backend.next_published()
+            if event is None:
+                # Backend is disconnected
+                break
+
             for queue in list(self._subscribers.get(event.channel, [])):
                 await queue.put(event)
 
+        # Ubsubscribe all
+        for queue in sum([list(qs) for qs in self._subscribers.values()], []):
```

Comment:
```python
# Unsubscribe all
for queue_set in self._subscribers.values():
    for queue in queue_set:
        await queue.put(None)
```
Seems like a lot less wasteful / clear what's going on.
As far as I know, you can loop over sets directly in python, no need
to convert to a list.